src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioUploadServiceImpl.java

```diff
@@ -67,9 +70,32 @@ public class AudioUploadServiceImpl implements AudioUploadService {
     private final MediaLinkedCaseRepository mediaLinkedCaseRepository;
     private final AudioAsyncService audioAsyncService;
 
+
     @Override
     public void addAudio(MultipartFile audioMultipartFile, AddAudioMetadataRequest addAudioMetadataRequest) {
+        String incomingChecksum;
+        try {
+            incomingChecksum = fileContentChecksum.calculate(audioMultipartFile.getInputStream());
+        } catch (IOException e) {
+            throw new DartsApiException(FAILED_TO_UPLOAD_AUDIO_FILE, "Failed to compute incoming checksum", e);
+        }
+        addAudio(incomingChecksum, () -> saveAudioToInbound(audioMultipartFile), addAudioMetadataRequest);
+    }
 
+    @Override
+    public void addAudio(UUID guid, AddAudioMetadataRequest addAudioMetadataRequest) {
+        String checksum = dataManagementApi.getChecksum(DatastoreContainerType.INBOUND, guid);
+        if (!checksum.equals(addAudioMetadataRequest.getChecksum())) {
+            throw new DartsApiException(AudioApiError.FAILED_TO_ADD_AUDIO_META_DATA,
+                                        String.format("Checksum in DETs (%s) does not match the one passed in the API request (%s).",
```

Comment:
Why are we talking about DETS here? It's nothing to do with DETS is it?

Reply:
This was left over from the first iteration I have updated the message :)